docs/extending/rich_text_internals.md

```diff
@@ -197,6 +197,31 @@ def register_embed_handler(features):
     features.register_embed_type(MyCustomEmbedHandler)
 ```
 
+### Registering generic rewrite handlers
+
+If you need to add a new rich text feature that does not fit the pattern of the LinkHandler or the EmbedHandler, you can register an entirely new rewriter alongside your converter rule.
+
+Rewriters must be callables, that receive a single argument - the rich-text HTML - and are registered with the `register_frontend_rewriter` call on the rich-text `features` object. By convention they are written as classes with a `__call__` method, but a function would also suffice for simple replacers.
+
+```python
+class GreenRewriter:
+    TAG_RE = re.compile(r"<green>(.+?)</green>")
+
+    def replace_tag(self, match):
+        content = match.groups(1)[0]
+        return '<span style="color: green">{content}</span>'.format(content=content)
```

Comment:
Could we use a class name instead? I’d prefer we avoid inline styles even just in documentation so we don’t add noise in our quest to remove all inline styles for CSP compatibility #1288).

Reply:
Done, in concert with below